src/core/toga/app.py

```diff
@@ -49,8 +50,7 @@ class App:
     app = None
 
     def __init__(self, name, app_id,
-                 id=None, icon=None, startup=None, document_types=None, factory=None):
-
+                 id=None, icon=None, startup=None, document_types=None, on_exit=None, factory=None):
```

Comment:
I guess `on_exit` is not really used in initialization here. But without it, if an app doesn't set `on_exit` handler, it is not initialized and an error is raised: App doesn't have `_on_exit`. Not sure what to do.

Reply:
Looks like you've almost got it - you're just missing one step. You've primed the `_on_exit` storage, which needs to happen before the call to `self.factory.App()`; then, after that call, you explicitly set the on_exit handler (i.e., around line 78, you call `self.on_exit = on_exit`).